### ISSUE TITLE
Support parsing HSQLDB CALL <expression> statement

### ISSUE THREAD

**Issue by jdsada30** [jOOQ/jOOQ#12002]
### SQL fails to translates this query:

```sql
CALL NEXT value FOR hibernate_sequence;
```

This statement is supported by HSQLDB dialect. Query comes from github project [AxonFramework ](https://github.com/AxonFramework/AxonFramework).

**Comment by lukaseder** [comment]
Thanks for your message. That's curious. I can find some references on the web about Hibernate doing it this way:
https://stackoverflow.com/a/2345488/521799

I can't seem to find authoritative documentation for this syntax (e.g. in http://hsqldb.org/doc/guide/sqlroutines-chapt.html), but looking at the parser implementation, it seems that `CALL` allows for evaluating any scalar expression, such as e.g.

```sql
CALL 1 + 1 
```

So, there isn't really a special syntax for sequences. It's just `CALL <expression>`, which we can obviously support in the parser, easily, translating it to `SELECT <expression>` if the expression isn't a stored function expression.

**Comment by lukaseder** [comment]
There's some ambiguity between dialects to parse arbitrary expressions. E.g. the above doesn't parse as `SELECT 1 + 1` yet, because in some dialects:

- `1` can be a valid identifier even without quoting
- routine invocations don't require parenthesised argument lists

So `CALL 1 + 1` might as well translate to `CALL "1" (+1)` (unary `+` operator).

**Comment by lukaseder** [comment]
This would be more helpful to disambiguate if the parser knew its language context (https://github.com/jOOQ/jOOQ/issues/11250). I.e. even in HSQLDB, the `CALL` statement works this way only if it's a top level statement. Otherwise, expressions seem not to be able to resolve themselves. E.g.

```sql
CREATE PROCEDURE p ()
BEGIN ATOMIC
  CALL 1 + 1;
END
```

Produces:

> SQL Error [42501]: user lacks privilege or object not found: 1

```sql
CREATE PROCEDURE p ()
BEGIN ATOMIC
  CALL NEXT VALUE FOR s;
END
```

Produces:

> SQL Error [42501]: user lacks privilege or object not found: NEXT

**Comment by lukaseder** [comment]
The disambiguation is easier if we implement this only for HSQLDB, which requires the parentheses around routine invocation argument lists (unless in Oracle compatibility mode).

**Comment by lukaseder** [comment]
H2 also supports this

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0. The online version on https://www.jooq.org/translate will be updated soon with the latest snapshot.

Thanks again for your report!
